class.jetpack.php

```diff
@@ -4104,6 +4105,14 @@ function build_connect_url( $raw = false, $redirect = false, $from = false, $reg
 
 		if ( $register || ! $token || ! $site_id ) {
 			$url = Jetpack::nonce_url_no_esc( Jetpack::admin_url( 'action=register' ), 'jetpack-register' );
+
+			if ( ! empty( $redirect ) ) {
```

Comment:
There's already a piece of code present in this function that deals with the redirect URL. What do you think about carrying it above the `if ( $register ...` condition instead of duplicating it here?